### DISCUSSION TITLE
I can't authorize my application using the provided documentation

### DISCUSSION THREAD

**Discussion by dorian-akhs** [tidal-music/discussions#29] [Q&A]
I'm trying to authorize my application using this [page](https://developer.tidal.com/documentation/concepts/concepts-authorization) information.
I've used a git bash console and could get Base64 creds using this command:
`B64CREDS=$(echo -n "CLIENT_ID:CLIENT_SECRET" | base64)`

However, when I execute the curl command (copy pasted), I get an HTTP Status 400 error:
`<!doctype html><html lang="en"><head><title>HTTP Status 400 – Bad Request</title><style type="text/css">body {font-family:Tahoma,Arial,sans-serif;} h1, h2, h3, b {color:white;background-color:#525D76;} h1 {font-size:22px;} h2 {font-size:16px;} h3 {font-size:14px;} p {font-size:12px;} a {color:black;} .line {height:1px;background-color:#525D76;border:none;}</style></head><body><h1>HTTP Status 400 – Bad Request</h1></body></html>`

I've also tried to execute this request using JS fetch(), but got a 400 status error as well.

Am I missing something ? The response error is in HTML/CSS which seems weird, is the documentation up to date ?

**Comment by stefanmoro** [comment]
Hi @dorian-akhs , thanks for reaching out.

Just to be clear -  in the `B64CREDS=$(echo -n "CLIENT_ID:CLIENT_SECRET" | base64)`, you have actually replaced the `CLIENT_ID` and `CLIENT_SECRET` with the corresponding values from your app?

I just tried this with multiple of my apps, and it works fine for me.

**Comment by dorian-akhs** [comment]
I found what's going on.

I'm indeed replacing `CLIENT_ID` and `CLIENT_SECRET` with the corresponding values from my app.
The issue is, when I'm running the command `B64CREDS=$(echo -n "CLIENT_ID:CLIENT_SECRET" | base64)` on my git bash terminal, the variable `B64CREDS` looks something like this: `falichnalkclakzhdc ldjalk==`, there is a space before the last 8 characters.

I tried using an [online base64 encoder](https://www.base64encode.org/), and I get no space in the output. By copy pasting the value I got with the online base64 encoder the curl request succeeds an I get my access tokens.

I just don't understand why my `B64CREDS=$(echo -n "CLIENT_ID:CLIENT_SECRET" | base64)` command doesn't give the same output as the one from the online base64 encoder.
